add validation for min and max on select field

aliceriot opened the pull request:
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?

closes #185 

#### What's this PR do?

this adds validation for the min and max property on a `Select` field (this is only enforced when the `multiple` prop is also set). I also added some general testing for the select validation, as there weren't yet any tests for it.

#### How should this be manually tested?

I made a little change to the `localdev/configs/ocw-course-site-config.yml` file, so you should be able to just run the `override_site_config` management command to add a multiple-select 'tag' field to the page content type. Then you can check in the UI that it's correctly enforcing the `min` and `max` (we may want to remove this again from `localdev/configs/ocw-course-site-config.yml` before we merge so it's not too annoying).

#### Screenshots (if appropriate)

![Screen Shot 2021-04-14 at 2 36 49 PM](https://user-images.githubusercontent.com/6207644/114761741-e6baf280-9d2e-11eb-885d-a44cbc58079f.png)
![Screen Shot 2021-04-14 at 2 37 02 PM](https://user-images.githubusercontent.com/6207644/114761742-e6baf280-9d2e-11eb-85bf-e500b9222296.png)

noisecapella commented:
If I start off with an empty tags list and click Save without modifying the field, it will save without a validation error. This probably won't happen with new pages but it could happen with existing pages that get a new field

aliceriot commented:
ok I fixed the issue, it related to not having a multiple-select specific default value, so an empty string was being put there instead of an empty list. I fixed it, so editing a page which does not have a value saved yet for `tags` should work (i.e. if you open the editor you should see a validation message saying 'tags must have at least one entry'
